nym-node-status-api/src/testruns/mod.rs

```diff
@@ -0,0 +1,83 @@
+use crate::db::models::GatewayIdentityDto;
+use crate::db::DbPool;
+use futures_util::TryStreamExt;
+use std::time::Duration;
+use tracing::instrument;
+
+pub(crate) mod models;
+mod queue;
+pub(crate) use queue::{now_utc, now_utc_as_rfc3339};
+
+// TODO dz should be configurable
+const REFRESH_DELAY: Duration = Duration::from_secs(60 * 5);
+
+pub(crate) async fn spawn(pool: DbPool) {
+    let pool = pool.clone();
+    tokio::spawn(async move {
+        // TODO dz delay for the first run, remove before merge
+        tokio::time::sleep(Duration::from_secs(90)).await;
+
+        loop {
+            tracing::info!("Spawning testruns...");
+
+            if let Err(e) = run(&pool).await {
+                tracing::error!("Cron job failed: {}", e);
+            }
+            tracing::debug!("Sleeping for {}s...", REFRESH_DELAY.as_secs());
+            tokio::time::sleep(REFRESH_DELAY).await;
+        }
+    });
+}
+
+// TODO dz make number of max agents configurable
+
+// TODO dz periodically clean up stale pending testruns
+#[instrument(level = "debug", name = "testrun_queue", skip_all)]
+async fn run(pool: &DbPool) -> anyhow::Result<()> {
+    if pool.is_closed() {
+        tracing::debug!("DB pool closed, returning early");
+        return Ok(());
+    }
+
+    let mut conn = pool.acquire().await?;
+
+    let gateways = sqlx::query_as!(
+        GatewayIdentityDto,
+        r#"SELECT
```

Comment:
can we keep all sql contained within a single file (or maybe to a traint)? I think this makes it easier to manage rather than finding random sql in the middle of an async task

Reply:
agree, will do